tla/consensus/ccfraft.tla

```diff
@@ -1080,7 +1082,8 @@ HandleAppendEntriesResponse(i, j, m) ==
           /\ leadershipState[i] = Leader \* Only Leaders need to tally append entries responses
           /\ m.success \* successful
           \* max(...) because why would we ever want to go backwards on a success response?!
-          /\ matchIndex' = [matchIndex EXCEPT ![i][j] = max(@, m.lastLogIndex)]
+          \* min(...) because we don't want to calculate a match past our own log's tail
+          /\ matchIndex' = [matchIndex EXCEPT ![i][j] = max(@, min(Len(log[i]), m.lastLogIndex))]
```

Comment:
Fixing a minor discrepancy between the spec and the implementation. Following the fixes above, we produce traces where this distinction actually matters, so update the spec to continue trace validation.

Reply:
Not sure I understand why this change is needed. A leader should never receive an AE-ACK in the current term with a index greater than its log

Reply:
I think you're right. The point where this is done in the implementation says `Potentially unnecessary safety check`. I tried removing both, but only in earlier versions that were failing TV or SIM for other reasons - it looks like this is now removable in both locations, so I'll try that.